x-pack/plugins/observability_solution/synthetics/public/apps/synthetics/components/settings/hooks/use_get_data_stream_statuses.ts

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useFetcher } from '@kbn/observability-shared-plugin/public';
+import { DataStream } from '@kbn/index-management-plugin/common';
+import { useContext } from 'react';
+import { getDslPolicies } from './api';
+import { SyntheticsRefreshContext } from '../../../contexts';
+import { formatBytes } from '../../step_details_page/hooks/use_object_metrics';
+import { formatAge } from '../data_retention/common';
+import { policyLabels } from '../data_retention/policy_labels';
+
+export type DataStreamStatus = DataStream & {
+  dataStreamName?: string;
+  isEmpty?: boolean;
+};
+
+export interface DataStreamStatusResponse {
+  dataStreamStatuses?: DataStreamStatus[];
+  error?: Error;
+  loading?: boolean;
+}
+
+/**
+ * Query the Data Streams API for stats and info about the current data streams.
+ *
+ * Filters out non-synthetics streams and overlays additional useful fields for display.
+ */
+export function useGetDataStreamStatuses(): DataStreamStatusResponse {
+  const { lastRefresh } = useContext(SyntheticsRefreshContext);
+  const { data, error, loading } = useFetcher(getDslPolicies, [lastRefresh]);
+
+  if (!Array.isArray(data) || !!error) return { dataStreamStatuses: undefined, error, loading };
+
+  const dataStreamSet = new Set();
+  const dataStreamMap: Record<string, DataStream> = {};
+
+  data.forEach((dataStream) => {
+    dataStreamSet.add(dataStream.indexTemplateName);
+    dataStreamMap[dataStream.indexTemplateName] = dataStream;
+  });
+
+  const dataStreamStatuses: DataStream[] = [];
+  let totalBytes = 0;
+  let summaryItem: DataStreamStatus | null = null;
+  for (const { indexTemplate, label } of policyLabels) {
+    if (!dataStreamSet.has(indexTemplate)) {
+      const missingStream = toMissingDataStream({ indexTemplate, label });
+      dataStreamStatuses.push(missingStream);
+      if (indexTemplate === 'synthetics') summaryItem = missingStream;
+    } else {
+      dataStreamStatuses.push(formatDataStreamInfo(dataStreamMap[indexTemplate]));
+      totalBytes += dataStreamMap[indexTemplate].storageSizeBytes ?? 0;
+    }
+  }
```

Comment:
What do you think of this? I didn't test it though.
```suggestion
  const dataStreamMap = keyBy(data, 'indexTemplateName');
  
  const dataStreamStatuses: DataStream[] = [];
  let totalBytes = 0;
  let summaryItem: DataStreamStatus | null = null;
  for (const { indexTemplate, label } of policyLabels) {
    const dataStream = dataStreamMap[indexTemplate]
    // JUSTIN -> I inverted the order here too, I think it's easy to follow when the truth logic is handled first.
    if (dataStream) {
      dataStreamStatuses.push(formatDataStreamInfo(dataStream));
      totalBytes += dataStream.storageSizeBytes ?? 0;
    } else {
      const missingStream = toMissingDataStream({ indexTemplate, label });
      dataStreamStatuses.push(missingStream);
      if (indexTemplate === 'synthetics') summaryItem = missingStream;
    }
  }
```

Reply:
I rarely use lodash but this does simplify it and remove a var so I'll follow this. 👍 

I inverted the logic as you recommended I agree it's easier to follow the positive.

1f597f37ecc87730597e6a04825fe97dab7d8d28